Add our label configuration

mcdonnnj opened the pull request:
<!-- GitHub renders PRs such that soft line breaks are treated as hard
line breaks.  In order to make this template render as expected we
therefore have to avoid soft breaks and therefore will offend
markdownlint with long lines.  This is the reason for the markdownline
disable directive just below.

For more details see:
https://github.com/github/markup/issues/1050#issuecomment-294654762 -->
<!-- markdownlint-disable MD013 -->
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request adds the label management and PR auto-labelling functionality that we use in our modern projects. I also made a few label updates that are project specific.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

This will help ensure that our usual labels are available in the project and also provide convenient auto-labelling of pull requests. This mirrors the work done in https://github.com/cisagov/cyhy-commander/pull/20.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

I eyeballed it. I also verified that the sync-label dry run would make the expected changes.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).

mcdonnnj committed:
Add our repository label configurations

Add the repository label management and pull request auto labeling
configurations from cisagov/skeleton-generic.

mcdonnnj committed:
Add setup.py to the `dependencies` label triggers

mcdonnnj committed:
Add a label for the package's command-line tools

Add a new label and auto-label configuration for the package's command-
line tools. Also add the command-line tool directory to the
configuration for the `python` label.

mcdonnnj committed:
Add `getenv` script to `shell script` label configuration

Since this is a shell script without a ".sh" extension it must be
manually configured to be properly labelled.

mcdonnnj commented on .github/workflows/label-prs.yml:
```diff
@@ -0,0 +1,93 @@
+          # this workflow.
+          permissions_monitoring_config: ${{ vars.ACTIONS_PERMISSIONS_CONFIG }}
+      - name: Apply suitable labels to a pull request
+        uses: actions/labeler@v6
```
I think we are ok using the default path without remarking such.

mcdonnnj commented on .github/workflows/sync-labels.yml:
```diff
@@ -0,0 +1,94 @@
+          # this workflow.
+          permissions_monitoring_config: ${{ vars.ACTIONS_PERMISSIONS_CONFIG }}
+      - uses: actions/checkout@v5
+      - name: Sync repository labels
+        if: success()
+        uses: crazy-max/ghaction-github-labeler@v5
```
I think we are ok using the default path without remarking such.

dav3r commented on .github/labels.yml:
```diff
@@ -0,0 +1,97 @@
+  description: This issue will be advertised on code.gov's Open Tasks page (https://code.gov/open-tasks)
+  name: code.gov
+- color: ffd343
+  description: Pull requests that update cyhy-core command-line tools
```
Correct repo name:
```suggestion
  description: Pull requests that update cyhy-reports command-line tools
```

mcdonnnj replied:
Ack, yep. I copied this from a cyhy-core branch and forgot to fix that.

mcdonnnj replied:
I went with your other suggestion in https://github.com/cisagov/cyhy-reports/pull/130/commits/aac6c16898ea668626a34ef6b1ca681d164c2843.

dav3r commented on .github/labels.yml:
```diff
@@ -0,0 +1,97 @@
+  description: This issue will be advertised on code.gov's Open Tasks page (https://code.gov/open-tasks)
+  name: code.gov
+- color: ffd343
+  description: Pull requests that update cyhy-core command-line tools
```
An alternative if we want one description to cover both cyhy-core and cyhy-reports:
```suggestion
  description: Pull requests that update CyHy command-line tools
```

mcdonnnj replied:
I think this is the better approach for conformity actually.

mcdonnnj committed:
Fix description for `command-line tools` label

Use a more universal reference to CyHy in general instead of a project-
specific reference.

Co-authored-by: dav3r <david.redmin@gwe.cisa.dhs.gov>
